fendermint/vm/interpreter/src/fvm/state/exec.rs

```diff
@@ -64,8 +80,11 @@ where
     /// execution interpreter without having to add yet another piece to track at the app level.
     block_hash: Option<BlockHash>,
 
-    /// Conversion between collateral and voting power.
-    power_scale: PowerScale,
+    /// State of parameters that are outside the control of the FVM but can change and need to be persisted.
+    params: FvmUpdatableParams,
+
+    /// Indicate whether the parameters have been updated.
+    params_dirty: bool,
```

Comment:
Just curious, I see this parameter updated but not read, is it used in other PRs?

Reply:
You are right, it's a bit obscure. The only reason I added it was to use it in [genesis](https://github.com/consensus-shipyard/fendermint/pull/330/files#diff-8f0901d357789002b95cef8fd68d18055c0ef0b06a3873ee9cea6396f177ea3cR155) where the `commit`, like the one here used to, returns just a `CID` but it's a unified one coming either from the `StateTree` or the `FvmExecState`. I didn't want to change that `commit` too, but I wanted to have a sanity check that there hasn't been some change that I am discarding, in which case this design would have to change too, e.g. by moving the fields from `FvmGenesisOutput` to the commit return value.

It's a bit convoluted, but I haven't found a nice way to do things. I tried with maintaining a copy of `FvmStateParams`, but not all those fields can change during a run, and for the ones that can be retrieved from the executor, there is always that ambiguity of which one to return. Now there is only one, because the `circ_supply` in the updatable part is write-only.